src/main/java/org/commcare/formplayer/util/FormplayerSentry.java

```diff
@@ -51,196 +40,128 @@
     @Autowired
     private RestoreFactory restoreFactory;
 
-    public FormplayerSentry(SentryClient sentryClient) {
-        this.sentryClient = sentryClient;
-    }
-
-    private void recordBreadcrumb(Breadcrumb breadcrumb) {
-        if (sentryClient == null) {
-            return;
-        }
-        try {
-            sentryClient.getContext().recordBreadcrumb(breadcrumb);
-        } catch (Exception e) {
-            log.info("Error recording breadcrumb. Ensure that sentryClient is configured. ", e);
-        }
-    }
-
-    public class BreadcrumbRecorder extends BreadcrumbBuilder {
-        FormplayerSentry parent;
+    public static class BreadcrumbRecorder {
+        Breadcrumb breadcrumb;
 
-        public BreadcrumbRecorder(FormplayerSentry parent) {
-            this.parent = parent;
+        public BreadcrumbRecorder() {
+            this.breadcrumb = new Breadcrumb();
         }
 
         public BreadcrumbRecorder setData(String... dataPairs) {
-            HashMap<String, String> data = new HashMap<>();
             // ignores last element if odd number of elements
             int length = dataPairs.length / 2;
             for (int i = 0; i < length; i++) {
-                data.put(dataPairs[2 * i], dataPairs[2 * i + 1]);
+                String key = dataPairs[2 * i];
+                String value = dataPairs[2 * i + 1];
+                if (key != null && value != null){
```

Comment:
nulls are not allowed

Reply:
Now that we have lombok integrated, can we set @nonnull on the param list to fail faster?

Reply:
yea, could do that

Reply:
I'm going to leave this as is since I can't tell if any of the elements will be null at compile time. Most of the data is coming from request beans so we should start with making those APIs clear in terms of what can and can't be null.